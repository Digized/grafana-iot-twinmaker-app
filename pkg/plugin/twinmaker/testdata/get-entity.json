{
    "Arn": "arn:aws:iottwinmaker:us-east-1:166800769179:workspace/AlarmWorkspace/entity/Mixer_1_4b57cbee-c391-4de6-b882-622c633a697e",
    "Components": {
        "AlarmComponent": {
            "ComponentName": "AlarmComponent",
            "ComponentTypeId": "com.example.cookiefactory.alarm",
            "DefinedIn": null,
            "Description": null,
            "Properties": {
                "alarm_key": {
                    "Definition": {
                        "Configuration": null,
                        "DataType": {
                            "AllowedValues": null,
                            "NestedType": null,
                            "Relationship": null,
                            "Type": "STRING",
                            "UnitOfMeasure": null
                        },
                        "DefaultValue": null,
                        "IsExternalId": true,
                        "IsFinal": false,
                        "IsImported": false,
                        "IsInherited": true,
                        "IsRequiredInEntity": true,
                        "IsStoredExternally": false,
                        "IsTimeSeries": false
                    },
                    "Value": {
                        "BooleanValue": null,
                        "DoubleValue": null,
                        "Expression": null,
                        "IntegerValue": null,
                        "ListValue": null,
                        "LongValue": null,
                        "MapValue": null,
                        "RelationshipValue": null,
                        "StringValue": "Mixer_1_597c735b-38fd-476c-b276-7592b1699ef8"
                    }
                },
                "alarm_status": {
                    "Definition": {
                        "Configuration": null,
                        "DataType": {
                            "AllowedValues": [
                                {
                                    "BooleanValue": null,
                                    "DoubleValue": null,
                                    "Expression": null,
                                    "IntegerValue": null,
                                    "ListValue": null,
                                    "LongValue": null,
                                    "MapValue": null,
                                    "RelationshipValue": null,
                                    "StringValue": "ACTIVE"
                                },
                                {
                                    "BooleanValue": null,
                                    "DoubleValue": null,
                                    "Expression": null,
                                    "IntegerValue": null,
                                    "ListValue": null,
                                    "LongValue": null,
                                    "MapValue": null,
                                    "RelationshipValue": null,
                                    "StringValue": "SNOOZE_DISABLED"
                                },
                                {
                                    "BooleanValue": null,
                                    "DoubleValue": null,
                                    "Expression": null,
                                    "IntegerValue": null,
                                    "ListValue": null,
                                    "LongValue": null,
                                    "MapValue": null,
                                    "RelationshipValue": null,
                                    "StringValue": "ACKNOWLEDGED"
                                },
                                {
                                    "BooleanValue": null,
                                    "DoubleValue": null,
                                    "Expression": null,
                                    "IntegerValue": null,
                                    "ListValue": null,
                                    "LongValue": null,
                                    "MapValue": null,
                                    "RelationshipValue": null,
                                    "StringValue": "NORMAL"
                                }
                            ],
                            "NestedType": null,
                            "Relationship": null,
                            "Type": "STRING",
                            "UnitOfMeasure": null
                        },
                        "DefaultValue": null,
                        "IsExternalId": false,
                        "IsFinal": false,
                        "IsImported": false,
                        "IsInherited": true,
                        "IsRequiredInEntity": false,
                        "IsStoredExternally": true,
                        "IsTimeSeries": true
                    },
                    "Value": null
                },
                "telemetryAssetId": {
                    "Definition": {
                        "Configuration": null,
                        "DataType": {
                            "AllowedValues": null,
                            "NestedType": null,
                            "Relationship": null,
                            "Type": "STRING",
                            "UnitOfMeasure": null
                        },
                        "DefaultValue": null,
                        "IsExternalId": false,
                        "IsFinal": false,
                        "IsImported": false,
                        "IsInherited": true,
                        "IsRequiredInEntity": true,
                        "IsStoredExternally": false,
                        "IsTimeSeries": false
                    },
                    "Value": {
                        "BooleanValue": null,
                        "DoubleValue": null,
                        "Expression": null,
                        "IntegerValue": null,
                        "ListValue": null,
                        "LongValue": null,
                        "MapValue": null,
                        "RelationshipValue": null,
                        "StringValue": "Mixer_1_597c735b-38fd-476c-b276-7592b1699ef8"
                    }
                },
                "telemetryAssetType": {
                    "Definition": {
                        "Configuration": null,
                        "DataType": {
                            "AllowedValues": null,
                            "NestedType": null,
                            "Relationship": null,
                            "Type": "STRING",
                            "UnitOfMeasure": null
                        },
                        "DefaultValue": {
                            "BooleanValue": null,
                            "DoubleValue": null,
                            "Expression": null,
                            "IntegerValue": null,
                            "ListValue": null,
                            "LongValue": null,
                            "MapValue": null,
                            "RelationshipValue": null,
                            "StringValue": "Alarm"
                        },
                        "IsExternalId": false,
                        "IsFinal": false,
                        "IsImported": false,
                        "IsInherited": true,
                        "IsRequiredInEntity": true,
                        "IsStoredExternally": false,
                        "IsTimeSeries": false
                    },
                    "Value": {
                        "BooleanValue": null,
                        "DoubleValue": null,
                        "Expression": null,
                        "IntegerValue": null,
                        "ListValue": null,
                        "LongValue": null,
                        "MapValue": null,
                        "RelationshipValue": null,
                        "StringValue": "Alarm"
                    }
                }
            },
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            }
        },
        "MixerComponent": {
            "ComponentName": "MixerComponent",
            "ComponentTypeId": "com.example.cookiefactory.mixer",
            "DefinedIn": null,
            "Description": null,
            "Properties": {
                "RPM": {
                    "Definition": {
                        "Configuration": null,
                        "DataType": {
                            "AllowedValues": null,
                            "NestedType": null,
                            "Relationship": null,
                            "Type": "DOUBLE",
                            "UnitOfMeasure": null
                        },
                        "DefaultValue": null,
                        "IsExternalId": false,
                        "IsFinal": false,
                        "IsImported": false,
                        "IsInherited": false,
                        "IsRequiredInEntity": false,
                        "IsStoredExternally": true,
                        "IsTimeSeries": true
                    },
                    "Value": null
                },
                "Temperature": {
                    "Definition": {
                        "Configuration": null,
                        "DataType": {
                            "AllowedValues": null,
                            "NestedType": null,
                            "Relationship": null,
                            "Type": "DOUBLE",
                            "UnitOfMeasure": null
                        },
                        "DefaultValue": null,
                        "IsExternalId": false,
                        "IsFinal": false,
                        "IsImported": false,
                        "IsInherited": false,
                        "IsRequiredInEntity": false,
                        "IsStoredExternally": true,
                        "IsTimeSeries": true
                    },
                    "Value": null
                },
                "telemetryAssetId": {
                    "Definition": {
                        "Configuration": null,
                        "DataType": {
                            "AllowedValues": null,
                            "NestedType": null,
                            "Relationship": null,
                            "Type": "STRING",
                            "UnitOfMeasure": null
                        },
                        "DefaultValue": null,
                        "IsExternalId": false,
                        "IsFinal": false,
                        "IsImported": false,
                        "IsInherited": true,
                        "IsRequiredInEntity": true,
                        "IsStoredExternally": false,
                        "IsTimeSeries": false
                    },
                    "Value": {
                        "BooleanValue": null,
                        "DoubleValue": null,
                        "Expression": null,
                        "IntegerValue": null,
                        "ListValue": null,
                        "LongValue": null,
                        "MapValue": null,
                        "RelationshipValue": null,
                        "StringValue": "Mixer_1_d2e1fb9c-308e-4801-9241-e4d3aa35255c"
                    }
                },
                "telemetryAssetType": {
                    "Definition": {
                        "Configuration": null,
                        "DataType": {
                            "AllowedValues": null,
                            "NestedType": null,
                            "Relationship": null,
                            "Type": "STRING",
                            "UnitOfMeasure": null
                        },
                        "DefaultValue": {
                            "BooleanValue": null,
                            "DoubleValue": null,
                            "Expression": null,
                            "IntegerValue": null,
                            "ListValue": null,
                            "LongValue": null,
                            "MapValue": null,
                            "RelationshipValue": null,
                            "StringValue": "Mixer"
                        },
                        "IsExternalId": false,
                        "IsFinal": false,
                        "IsImported": false,
                        "IsInherited": true,
                        "IsRequiredInEntity": true,
                        "IsStoredExternally": false,
                        "IsTimeSeries": false
                    },
                    "Value": {
                        "BooleanValue": null,
                        "DoubleValue": null,
                        "Expression": null,
                        "IntegerValue": null,
                        "ListValue": null,
                        "LongValue": null,
                        "MapValue": null,
                        "RelationshipValue": null,
                        "StringValue": "Mixer"
                    }
                }
            },
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            }
        },
        "SpecSheets": {
            "ComponentName": "SpecSheets",
            "ComponentTypeId": "com.amazon.iottwinmaker.documents",
            "DefinedIn": null,
            "Description": null,
            "Properties": {
                "documents": {
                    "Definition": {
                        "Configuration": null,
                        "DataType": {
                            "AllowedValues": null,
                            "NestedType": {
                                "AllowedValues": null,
                                "NestedType": null,
                                "Relationship": null,
                                "Type": "STRING",
                                "UnitOfMeasure": null
                            },
                            "Relationship": null,
                            "Type": "MAP",
                            "UnitOfMeasure": null
                        },
                        "DefaultValue": null,
                        "IsExternalId": false,
                        "IsFinal": false,
                        "IsImported": false,
                        "IsInherited": false,
                        "IsRequiredInEntity": false,
                        "IsStoredExternally": false,
                        "IsTimeSeries": false
                    },
                    "Value": {
                        "BooleanValue": null,
                        "DoubleValue": null,
                        "Expression": null,
                        "IntegerValue": null,
                        "ListValue": null,
                        "LongValue": null,
                        "MapValue": {
                            "manufacturer_page": {
                                "BooleanValue": null,
                                "DoubleValue": null,
                                "Expression": null,
                                "IntegerValue": null,
                                "ListValue": null,
                                "LongValue": null,
                                "MapValue": null,
                                "RelationshipValue": null,
                                "StringValue": "https://console.aws.amazon.com/iottwinmaker/home"
                            },
                            "mixer1_manual": {
                                "BooleanValue": null,
                                "DoubleValue": null,
                                "Expression": null,
                                "IntegerValue": null,
                                "ListValue": null,
                                "LongValue": null,
                                "MapValue": null,
                                "RelationshipValue": null,
                                "StringValue": "https://github.com/aws-samples/aws-iot-twinmaker-samples"
                            }
                        },
                        "RelationshipValue": null,
                        "StringValue": null
                    }
                }
            },
            "Status": {
                "Error": {
                    "Code": null,
                    "Message": null
                },
                "State": "ACTIVE"
            }
        }
    },
    "CreationDateTime": "2022-04-27T17:29:56.17Z",
    "Description": "",
    "EntityId": "Mixer_1_4b57cbee-c391-4de6-b882-622c633a697e",
    "EntityName": "Mixer_1",
    "HasChildEntities": false,
    "ParentEntityId": "Mixers_b1e081ee-e6a3-4636-82cb-6b6bce0786a7",
    "Status": {
        "Error": {
            "Code": null,
            "Message": null
        },
        "State": "ACTIVE"
    },
    "UpdateDateTime": "2022-04-27T17:30:25.319Z",
    "WorkspaceId": "AlarmWorkspace"
}