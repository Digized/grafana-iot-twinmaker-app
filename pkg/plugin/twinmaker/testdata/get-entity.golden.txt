ðŸŒŸ This was machine generated.  Do not edit. ðŸŒŸ

Frame[0] {
    "custom": {}
}
Name: Mixer_1
Dimensions: 5 Fields by 3 Rows
+-----------------+-----------------------------------+-------------------+-------------------------------------------------------+-----------------------+
| Name: component | Name: componentTypeId             | Name: description | Name: properties                                      | Name: timeSeries      |
| Labels:         | Labels:                           | Labels:           | Labels:                                               | Labels:               |
| Type: []string  | Type: []string                    | Type: []*string   | Type: []string                                        | Type: []string        |
+-----------------+-----------------------------------+-------------------+-------------------------------------------------------+-----------------------+
| AlarmComponent  | com.example.cookiefactory.alarm   | null              | ["alarm_key","telemetryAssetId","telemetryAssetType"] | ["alarm_status"]      |
| MixerComponent  | com.example.cookiefactory.mixer   | null              | ["telemetryAssetId","telemetryAssetType"]             | ["RPM","Temperature"] |
| SpecSheets      | com.amazon.iottwinmaker.documents | null              | ["documents"]                                         | []                    |
+-----------------+-----------------------------------+-------------------+-------------------------------------------------------+-----------------------+


====== TEST DATA RESPONSE (arrow base64) ======
FRAME=QVJST1cxAAD/////eAMAABAAAAAAAAoADgAMAAsABAAKAAAAFAAAAAAAAAEEAAoADAAAAAgABAAKAAAACAAAAIQAAAADAAAAUAAAACgAAAAEAAAAGP3//wgAAAAMAAAAAAAAAAAAAAAFAAAAcmVmSWQAAAA4/f//CAAAABAAAAAHAAAATWl4ZXJfMQAEAAAAbmFtZQAAAABc/f//CAAAABgAAAANAAAAeyJjdXN0b20iOnt9fQAAAAQAAABtZXRhAAAAAAUAAABUAgAA1AEAAHABAACwAAAABAAAANL9//8UAAAAiAAAAIgAAAAAAAAFhAAAAAIAAAAwAAAABAAAAMT9//8IAAAAFAAAAAoAAAB0aW1lU2VyaWVzAAAEAAAAbmFtZQAAAADs/f//CAAAADAAAAAmAAAAeyJjdXN0b20iOnsiZGlzcGxheU1vZGUiOiJqc29uLXZpZXcifX0AAAYAAABjb25maWcAAAAAAAAA/v//CgAAAHRpbWVTZXJpZXMAAHr+//8UAAAAiAAAAIgAAAAAAAAFhAAAAAIAAAAwAAAABAAAAGz+//8IAAAAFAAAAAoAAABwcm9wZXJ0aWVzAAAEAAAAbmFtZQAAAACU/v//CAAAADAAAAAmAAAAeyJjdXN0b20iOnsiZGlzcGxheU1vZGUiOiJqc29uLXZpZXcifX0AAAYAAABjb25maWcAAAAAAACo/v//CgAAAHByb3BlcnRpZXMAAAAAEgAYABQAEwASAAwAAAAIAAQAEgAAABQAAABAAAAAQAAAAAAABQE8AAAAAQAAAAQAAAAk////CAAAABQAAAALAAAAZGVzY3JpcHRpb24ABAAAAG5hbWUAAAAAAAAAABz///8LAAAAZGVzY3JpcHRpb24Alv///xQAAABEAAAARAAAAAAAAAVAAAAAAQAAAAQAAACE////CAAAABgAAAAPAAAAY29tcG9uZW50VHlwZUlkAAQAAABuYW1lAAAAAAAAAACA////DwAAAGNvbXBvbmVudFR5cGVJZAAAABIAGAAUAAAAEwAMAAAACAAEABIAAAAUAAAASAAAAEwAAAAAAAAFSAAAAAEAAAAMAAAACAAMAAgABAAIAAAACAAAABQAAAAJAAAAY29tcG9uZW50AAAABAAAAG5hbWUAAAAAAAAAAAQABAAEAAAACQAAAGNvbXBvbmVudAAAAP////+YAQAAFAAAAAAAAAAMABYAFAATAAwABAAMAAAAeAEAAAAAAAAUAAAAAAAAAwQACgAYAAwACAAEAAoAAAAUAAAACAEAAAMAAAAAAAAAAAAAAA8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAABAAAAAAAAAAJgAAAAAAAAA4AAAAAAAAAAAAAAAAAAAAOAAAAAAAAAAQAAAAAAAAAEgAAAAAAAAAXwAAAAAAAACoAAAAAAAAAAEAAAAAAAAAsAAAAAAAAAAQAAAAAAAAAMAAAAAAAAAAAAAAAAAAAADAAAAAAAAAAAAAAAAAAAAAwAAAAAAAAAAQAAAAAAAAANAAAAAAAAAAawAAAAAAAABAAQAAAAAAAAAAAAAAAAAAQAEAAAAAAAAQAAAAAAAAAFABAAAAAAAAJwAAAAAAAAAAAAAABQAAAAMAAAAAAAAAAAAAAAAAAAADAAAAAAAAAAAAAAAAAAAAAwAAAAAAAAADAAAAAAAAAAMAAAAAAAAAAAAAAAAAAAADAAAAAAAAAAAAAAAAAAAAAAAAAA4AAAAcAAAAJgAAAEFsYXJtQ29tcG9uZW50TWl4ZXJDb21wb25lbnRTcGVjU2hlZXRzAAAAAAAAHwAAAD4AAABfAAAAY29tLmV4YW1wbGUuY29va2llZmFjdG9yeS5hbGFybWNvbS5leGFtcGxlLmNvb2tpZWZhY3RvcnkubWl4ZXJjb20uYW1hem9uLmlvdHR3aW5tYWtlci5kb2N1bWVudHMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADUAAABeAAAAawAAAFsiYWxhcm1fa2V5IiwidGVsZW1ldHJ5QXNzZXRJZCIsInRlbGVtZXRyeUFzc2V0VHlwZSJdWyJ0ZWxlbWV0cnlBc3NldElkIiwidGVsZW1ldHJ5QXNzZXRUeXBlIl1bImRvY3VtZW50cyJdAAAAAAAAAAAAEAAAACUAAAAnAAAAWyJhbGFybV9zdGF0dXMiXVsiUlBNIiwiVGVtcGVyYXR1cmUiXVtdABAAAAAMABQAEgAMAAgABAAMAAAAEAAAACwAAAA8AAAAAAAEAAEAAACIAwAAAAAAAKABAAAAAAAAeAEAAAAAAAAAAAAAAAAAAAAAAAAAAAoADAAAAAgABAAKAAAACAAAAIQAAAADAAAAUAAAACgAAAAEAAAAGP3//wgAAAAMAAAAAAAAAAAAAAAFAAAAcmVmSWQAAAA4/f//CAAAABAAAAAHAAAATWl4ZXJfMQAEAAAAbmFtZQAAAABc/f//CAAAABgAAAANAAAAeyJjdXN0b20iOnt9fQAAAAQAAABtZXRhAAAAAAUAAABUAgAA1AEAAHABAACwAAAABAAAANL9//8UAAAAiAAAAIgAAAAAAAAFhAAAAAIAAAAwAAAABAAAAMT9//8IAAAAFAAAAAoAAAB0aW1lU2VyaWVzAAAEAAAAbmFtZQAAAADs/f//CAAAADAAAAAmAAAAeyJjdXN0b20iOnsiZGlzcGxheU1vZGUiOiJqc29uLXZpZXcifX0AAAYAAABjb25maWcAAAAAAAAA/v//CgAAAHRpbWVTZXJpZXMAAHr+//8UAAAAiAAAAIgAAAAAAAAFhAAAAAIAAAAwAAAABAAAAGz+//8IAAAAFAAAAAoAAABwcm9wZXJ0aWVzAAAEAAAAbmFtZQAAAACU/v//CAAAADAAAAAmAAAAeyJjdXN0b20iOnsiZGlzcGxheU1vZGUiOiJqc29uLXZpZXcifX0AAAYAAABjb25maWcAAAAAAACo/v//CgAAAHByb3BlcnRpZXMAAAAAEgAYABQAEwASAAwAAAAIAAQAEgAAABQAAABAAAAAQAAAAAAABQE8AAAAAQAAAAQAAAAk////CAAAABQAAAALAAAAZGVzY3JpcHRpb24ABAAAAG5hbWUAAAAAAAAAABz///8LAAAAZGVzY3JpcHRpb24Alv///xQAAABEAAAARAAAAAAAAAVAAAAAAQAAAAQAAACE////CAAAABgAAAAPAAAAY29tcG9uZW50VHlwZUlkAAQAAABuYW1lAAAAAAAAAACA////DwAAAGNvbXBvbmVudFR5cGVJZAAAABIAGAAUAAAAEwAMAAAACAAEABIAAAAUAAAASAAAAEwAAAAAAAAFSAAAAAEAAAAMAAAACAAMAAgABAAIAAAACAAAABQAAAAJAAAAY29tcG9uZW50AAAABAAAAG5hbWUAAAAAAAAAAAQABAAEAAAACQAAAGNvbXBvbmVudAAAAKgDAABBUlJPVzE=
